packages/caver-wallet/src/keyringContainer.js

```diff
@@ -0,0 +1,301 @@
+/*
+    Copyright 2020 The caver-js Authors
+    This file is part of the caver-js library.
+
+    The caver-js library is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    The caver-js library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+    GNU Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    along with the caver-js. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+const _ = require('lodash')
+const Keyring = require('./keyring/keyring')
+const TransactionHasher = require('../../caver-transaction/src/transactionHasher/transactionHasher')
+const { KEY_ROLE } = require('./keyring/keyringHelper')
+const utils = require('../../caver-utils/src')
+
+/**
+ * representing a Keyring container which manages keyrings.
+ * @class
+ */
+class KeyringContainer {
+    /**
+     * creates a keyringContainer.
+     * @param {Array.<Keyring>} keyrings - The keyrings to be managed in KeyringContainer.
+     */
+    constructor(keyrings) {
+        keyrings = keyrings || []
+        this._addressKeyringMap = new Map()
+
+        // add keyrings to keyringContainer
+        for (const keyring of keyrings) {
+            this.add(keyring)
+        }
+    }
+
+    /**
+     * @type {number}
+     */
+    get length() {
+        return this._addressKeyringMap.size
+    }
+
+    /**
+     * generates keyrings in the keyringContainer with randomly generated key pairs.
+     *
+     * @param {number} numberOfKeyrings The number of accounts to create.
+     * @param {string} [entropy] A random string to increase entropy. If undefined, a random string will be generated using randomHex.
+     * @return {KeyringContainer}
+     */
+    generate(numberOfKeyrings, entropy) {
+        for (let i = 0; i < numberOfKeyrings; ++i) {
+            this.add(Keyring.generate(entropy))
+        }
+        return this
+    }
+
+    /**
+     * creates a keyring instance with given parameters and adds it to the keyringContainer.
```

Comment:
I think it would be good to add a description of the KeyringContainer mapping structure <Key, Value>.
Key is Address
Value is Keyring Object

Reply:
I added description for that :) Please take a look!